python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,514 @@
+    # create the new extented source catalog
+    srcVis = SourceCatalog(newSchema)
+
+    for vId in visitDataIds:
+        calib = afwImage.Calib(butler.get("calexp_md", vId, immediate=True))
+        calib.setThrowOnNegativeFlux(False)
+        oldSrc = butler.get('src', vId, immediate=True)
+        print(len(oldSrc), "sources in ccd: ", vId[ccdKeyName])
+
+        # create temporary catalog
+        tmpCat = SourceCatalog(SourceCatalog(newSchema).table)
+        tmpCat.extend(oldSrc, mapper=mapper)
+        (tmpCat['base_PsfFlux_mag'][:],
+         tmpCat['base_PsfFlux_magerr'][:]) = \
+             calib.getMagnitude(tmpCat['base_PsfFlux_flux'],
+                                tmpCat['base_PsfFlux_fluxSigma'])
+        srcVis.extend(tmpCat, False)
+        mmatch.add(catalog=tmpCat, dataId=vId)
+
+    # Complete the match, returning a catalog that includes all matched sources with object IDs that can be used to group them.
+    matchCat = mmatch.finish()
+
+    # Create a mapping object that allows the matches to be manipulated as a mapping of object ID to catalog of sources.
+    allMatches = GroupView.build(matchCat)
+    
+    return allMatches
+
+
+def analyzeData(allMatches, good_mag_limit=19.5):
+    """Calculate summary statistics for each star.
+
+    Inputs
+    ------
+    allMatches : afw.table.GroupView 
+        GroupView object with matches.
+    good_mag_limit : float, optional
+        Minimum average brightness (in magnitudes) for a star to be considered.
+
+    Returns
+    ------- 
+    pipeBase.Struct 
+        Containing mag, magerr, magrms, dist, and number of matches.
+    """
+
+    # Filter down to matches with at least 2 sources and good flags
+    flagKeys = [allMatches.schema.find("base_PixelFlags_flag_%s" % flag).key
+                for flag in ("saturated", "cr", "bad", "edge")]
+    nMatchesRequired = 2
+
+    psfMagKey = allMatches.schema.find("base_PsfFlux_mag").key
+    psfMagErrKey = allMatches.schema.find("base_PsfFlux_magerr").key
+    extendedKey = allMatches.schema.find("base_ClassificationExtendedness_value").key
+
+    def goodFilter(cat):
+        if len(cat) < nMatchesRequired:
+            return False
+        for flagKey in flagKeys:
+            if cat.get(flagKey).any():
+                return False
+        if not np.isfinite(cat.get(psfMagKey)).all():
+            return False
+        return True
+
+    goodMatches = allMatches.where(goodFilter)
+
+    # Filter further to a limited range in magnitude and extendedness 
+    # to select bright stars.  
+    safeMaxMag = good_mag_limit
+    safeMaxExtended = 1.0
+
+    def safeFilter(cat):
+        psfMag = np.mean(cat.get(psfMagKey))
+        extended = np.max(cat.get(extendedKey))
+        return psfMag <= safeMaxMag and extended < safeMaxExtended
+
+    safeMatches = goodMatches.where(safeFilter)
+
+    # Pass field=psfMagKey so np.mean just gets that as its input
+    goodPsfMag = goodMatches.aggregate(np.mean, field=psfMagKey)  # mag
```

Comment:
Not sure what the significance of the `# mag` is -- it seems clear this should be a magnitude without this.